packages/libs/client/src/signing/eckoWallet/eckoCommon.ts

```diff
@@ -0,0 +1,46 @@
+import { IEckoConnectOrStatusResponse } from './eckoTypes';
+
+export const isInstalled = (): boolean => {
+  const { kadena } = window;
+  return Boolean(kadena && kadena.isKadena);
+};
+
+export const isConnected = async (networkId: string): Promise<boolean> => {
+  if (!isInstalled()) {
+    return false;
+  }
+
+  const checkStatusResponse =
+    await window.kadena?.request<IEckoConnectOrStatusResponse>({
+      method: 'kda_checkStatus',
+      networkId,
+    });
+
+  if (checkStatusResponse?.status === 'fail') {
+    return false;
+  }
+
+  return true;
+};
+
+export const connect = async (networkId: string): Promise<boolean> => {
+  if (!isInstalled()) {
+    throw new Error('Ecko Wallet is not installed');
+  }
+
+  if (await isConnected(networkId)) {
+    return true;
+  }
+
+  const connectResponse =
+    await window.kadena?.request<IEckoConnectOrStatusResponse>({
+      method: 'kda_connect',
+      networkId,
+    });
```

Comment:
Does this ever throw?

Reply:
Only when `request` doesn't exist on `window.kadena`, which is checked in `isConnected`. I've now added an extra check in that function to check if `window.kadena.request` is there.